src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -7019,20 +7019,32 @@ public override IOperation VisitSwitchExpression(ISwitchExpressionOperation oper
 
             LeaveRegionsUpTo(resultCaptureRegion);
 
-            // throw new SwitchExpressionException
-            var matchFailureCtor =
-                (IMethodSymbol?)(_compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_SwitchExpressionException__ctor) ??
-                                _compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_InvalidOperationException__ctor))?.GetISymbol();
-            var makeException = (matchFailureCtor is null)
-                ? MakeInvalidOperation(operation.Syntax, type: _compilation.GetSpecialType(SpecialType.System_Object), ImmutableArray<IOperation>.Empty)
-                : new ObjectCreationOperation(
-                    matchFailureCtor, initializer: null, ImmutableArray<IArgumentOperation>.Empty, semanticModel: null, operation.Syntax,
-                    type: matchFailureCtor.ContainingType, constantValue: null, isImplicit: true);
-            LinkThrowStatement(makeException);
+            bool linkToPreviousAfterSwitch;
+            if (!operation.IsExhaustive)
+            {
+                // throw new SwitchExpressionException
+                var matchFailureCtor =
+                    (IMethodSymbol?)(_compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_SwitchExpressionException__ctor) ??
+                                    _compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_InvalidOperationException__ctor))?.GetISymbol();
+                var makeException = (matchFailureCtor is null)
+                    ? MakeInvalidOperation(operation.Syntax, type: _compilation.GetSpecialType(SpecialType.System_Object), ImmutableArray<IOperation>.Empty)
+                    : new ObjectCreationOperation(
+                        matchFailureCtor, initializer: null, ImmutableArray<IArgumentOperation>.Empty, semanticModel: null, operation.Syntax,
+                        type: matchFailureCtor.ContainingType, constantValue: null, isImplicit: true);
+                LinkThrowStatement(makeException);
+                AppendNewBlock(afterSwitch, linkToPrevious: true);
+
+                linkToPreviousAfterSwitch = true;
```

Comment:
>linkToPreviousAfterSwitch [](http://example.com/codeflow?start=16&length=25)

I think this should be the other way around. If we get here, the ```AppendNewBlock(afterSwitch, linkToPrevious: linkToPreviousAfterSwitch);``` should pass false as the second argument (this is how it used to be before adition of the `if`). If we get into the `else` part, we should pass true as the second argument. Also, consider if the local is needed, it looks like we could simply pass ```operation.IsExhaustive``` as the second argument.

Reply:
Could you change the code? I have no idea what "AppendNewBlock" does or should do...